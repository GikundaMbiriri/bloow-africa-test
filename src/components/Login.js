import React,{useEffect,useState} from 'react'
import { UserAddIcon } from '@heroicons/react/solid'
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { loginUser } from "../redux/actions/userAction";
import PropTypes from "prop-types";
import { withRouter } from 'react-router-dom';
function Login(props) {

    const [email,setEmail]=useState('');
    const [password,setPassword]=useState('');

    const handleSubmit=()=>{
      

        const userData = {
          email,
          password
        };
        console.log(userData)
        props.loginUser(userData, props.history);
    }
  return (
    <>
<div className='w-4/5 md:w-1/3 shadow-2xl m-auto mb-10 mt-5 py-5'>
<UserAddIcon className="h-12 w-12 mx-auto"/>
<h1 className='text-2xl font-bold pb-3 text-center text-dark-blue'>Login Here</h1>
<h1 className='text-center text-lg'>Remember Everything important</h1>
<div className="flex flex-col justify-center ml-8">
  <div className="mb-3 xl:w-96">
    <label for="exampleEmail0" className="form-label inline-block mb-2 text-gray-700"
      >Email input</label
    >
    <input
      type="email"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      id="exampleEmail0"
      value={email}
onChange={(event)=>setEmail(event.target.value)}
      placeholder="Email input"
    />
  </div>
  <div className="mb-3 xl:w-96">
    <label for="examplePassword0" className="form-label inline-block mb-2 text-gray-700"
      >Password input</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={password}
      onChange={(event)=>setPassword(event.target.value)}
      id="examplePassword0"
      placeholder="Password input"
    />
  </div>
  <div className="flex space-x-2 justify-center mb-3">
  <button type="button" className="inline-block px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out" onClick={handleSubmit}>Login</button>
</div>

</div>
<div className='mb-3'>
<h1 className='text-center'>Don't have an account ?</h1>
<Link to="/register">
<h1 className='text-center text-blue-600 text-sm'>Register here</h1>
</Link>
</div>
</div>
    </>
  )
}

Login.propTypes = {
  
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  UI: PropTypes.object.isRequired,
};
const mapStateToProps = (state) => ({
  user: state.user,
  UI: state.UI,
});
const mapActionsToProps = {
  loginUser,
};
export default connect(
  mapStateToProps,
  mapActionsToProps
)(withRouter(Login));