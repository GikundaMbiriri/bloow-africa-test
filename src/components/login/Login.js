import React,{useEffect,useState} from 'react'
import { UserAddIcon } from '@heroicons/react/solid'
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { loginUser } from "../../redux/actions/userAction";
import PropTypes from "prop-types";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import { withRouter } from 'react-router-dom';
function Login(props) {

    const [email,setEmail]=useState('');
    const [password,setPassword]=useState('');
const {loading,errors}=props.UI;
const Loading = () => {
    let circleCommonClasses = 'h-2.5 w-2.5 bg-current  rounded-full';

    return (
        <div className='flex justify-center py-2'>
            <div className={`${circleCommonClasses} mr-1 animate-bounce`}></div>
            <div
                className={`${circleCommonClasses} mr-1 animate-bounce200`}
            ></div>
            <div className={`${circleCommonClasses} animate-bounce400`}></div>
        </div>
    );
};
    const handleSubmit=()=>{
      

        const userData = {
          email,
          password
        };
        console.log(userData)
        props.loginUser(userData, props.history);
    }
  return (
    <>
<div className='w-4/5 md:w-1/3 shadow-2xl m-auto mb-10 mt-5 py-5'>
<UserAddIcon className="h-12 w-12 mx-auto"/>
<h1 className='text-2xl font-bold pb-3 text-center text-dark-blue'>Login Here</h1>
<h1 className='text-center text-lg' data-testid='egeneral'>{errors?<span className='text-center text-md text-red-600' >{errors.general}</span>:<span>Complete this form correctly</span>}</h1>
<div className="flex flex-col justify-center ml-8">
  <div className="mb-3 xl:w-96">
    <label htmlFor="exampleEmail0" className="form-label inline-block mb-2 text-gray-700"
      >Email input</label
    >
    <input
      type="email"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      id="exampleEmail0"
      data-testid="emailId"
      value={email}
onChange={(event)=>setEmail(event.target.value)}
      placeholder="Email input"
      
    />
          <h2 className='text-center text-md text-red-600' data-testid='eemail'>{errors?errors.email:''}</h2>

  </div>
  <div className="mb-3 xl:w-96">
    <label htmlFor="examplePassword0" className="form-label inline-block mb-2 text-gray-700"
      >Password input</label
    >
    <input
      type="password"
      className="
        form-control
        block
        w-full
        px-3
        py-1.5
        text-base
        font-normal
        text-gray-700
        bg-white bg-clip-padding
        border border-solid border-gray-300
        rounded
        transition
        ease-in-out
        m-0
        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
      "
      value={password}
      data-testid='passwordId'
      onChange={(event)=>setPassword(event.target.value)}
      id="examplePassword0"
      placeholder="Password input"
    />
              <h2 className='text-center text-md text-red-600' data-testid='epassword'>{errors?errors.password:''}</h2>

  </div>
  <div className="flex space-x-2 justify-center mb-3">

  <button type="button" className="inline-block px-6 py-2.5 bg-blue-600
   text-white font-medium text-xs leading-tight uppercase rounded shadow-md 
   hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none
    focus:ring-0 active:bg-blue-800 active:shadow-lg 
    transition duration-150 ease-in-out" onClick={handleSubmit} data-testid='loginid'>{loading?<Loading data-testid='loading1'/>:<span>Login</span>}</button>
</div>

</div>
<div className='mb-3'>
<h1 className='text-center'>Don't have an account ?</h1>
<Link to="/register">
<h1 className='text-center text-blue-600 text-sm'>Register here</h1>
</Link>
</div>
</div>
    </>
  )
}

Login.propTypes = {
  
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  UI: PropTypes.object.isRequired,
};
const mapStateToProps = (state) => ({
  user: state.user,
  UI: state.UI,
});
const mapActionsToProps = {
  loginUser,
};
export default connect(
  mapStateToProps,
  mapActionsToProps
)(withRouter(Login));
