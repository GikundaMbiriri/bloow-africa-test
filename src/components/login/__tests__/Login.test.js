import React from 'react'
import ReactDOM from 'react-dom'
import Login from '../Login';
import {render as rtlRender, screen} from '@testing-library/react'
import renderer from 'react-test-renderer';
import store from '../../../redux/store';
import {Provider} from 'react-redux';
import { HashRouter as Router } from "react-router-dom";

const render=component=>rtlRender(
    <Provider store={store}>
        <Router>
        {component}
        </Router>
    </Provider>
)
it("it renders without crashing",()=>{
    const div = document.createElement('div');
    render(<Login></Login>,div)
})

it('renders correctly', () => {
    const tree = renderer
      .create(
      <Provider store={store}>
          <Router>
      <Login></Login>
      </Router>
      </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('Login',()=>{
    test('input fields should be filled correctly',()=>{
        render(<Login/>)
        const credentials = { email: 'test@gmail.com', password: 'testpass' };
        const emailInput=screen.getByTestId('emailId');
        emailInput.value=credentials.email;
        const passwordInput=screen.getByTestId('passwordId');
        passwordInput.value=credentials.password;
     

        expect(emailInput.value).toBe('test@gmail.com');
        expect(passwordInput.value).toBe('testpass');
    
    })
    test('the login button displays login text so long as the component is not loading',()=>{
       let state = store.getState();

        render(<Login/>)
    if(!state.UI.loading){
        expect(screen.getByTestId('loginid')).toHaveTextContent('Login')

    }




     
    
    })
    test('the login button does does not display the login text when the component is loading',()=>{
        let state = store.getState();
        state.UI['loading']=true;
        render(<Login/>)
            expect(screen.getByTestId('loginid')).not.toHaveTextContent('Login')
        
        
    })
    test('Error messages do not display when there are no errors',()=>{
        let state = store.getState();
        render(<Login/>)
     
        if(!state.UI.errors){
            expect(screen.getByTestId('epassword')).toHaveTextContent('');
            expect(screen.getByTestId('eemail')).toHaveTextContent('');
            expect(screen.getByTestId('egeneral')).toHaveTextContent('Complete this form correctly')

        }
 
    })
    test('displays password error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={password:'Password field cannot be empty'};

        render(<Login/>) 
        expect(screen.getByTestId('epassword')).toHaveTextContent('Password field cannot be empty');

     })
    test('displays email error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={email:'Email field cannot be empty'};

        render(<Login/>) 
        expect(screen.getByTestId('eemail')).toHaveTextContent('Email field cannot be empty');

     })
    test('displays general error message correctly', () => {
        let state = store.getState();
        state.UI['errors']={general:'General field cannot be empty'};

        render(<Login/>) 
        expect(screen.getByTestId('egeneral')).toHaveTextContent('General field cannot be empty');

     })

})