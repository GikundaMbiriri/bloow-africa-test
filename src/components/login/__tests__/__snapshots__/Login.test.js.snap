// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-4/5 md:w-1/3 shadow-2xl m-auto mb-10 mt-5 py-5"
>
  <svg
    aria-hidden="true"
    className="h-12 w-12 mx-auto"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
    />
  </svg>
  <h1
    className="text-2xl font-bold pb-3 text-center text-dark-blue"
  >
    Login Here
  </h1>
  <h1
    className="text-center text-lg"
    data-testid="egeneral"
  >
    <span>
      Complete this form correctly
    </span>
  </h1>
  <div
    className="flex flex-col justify-center ml-8"
  >
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="exampleEmail0"
      >
        Email input
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="emailId"
        id="exampleEmail0"
        onChange={[Function]}
        placeholder="Email input"
        type="email"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="eemail"
      >
        
      </h2>
    </div>
    <div
      className="mb-3 xl:w-96"
    >
      <label
        className="form-label inline-block mb-2 text-gray-700"
        htmlFor="examplePassword0"
      >
        Password input
      </label>
      <input
        className="
 form-control
 block
 w-full
 px-3
 py-1.5
 text-base
 font-normal
 text-gray-700
 bg-white bg-clip-padding
 border border-solid border-gray-300
 rounded
 transition
 ease-in-out
 m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none
 "
        data-testid="passwordId"
        id="examplePassword0"
        onChange={[Function]}
        placeholder="Password input"
        type="password"
        value=""
      />
      <h2
        className="text-center text-md text-red-600"
        data-testid="epassword"
      >
        
      </h2>
    </div>
    <div
      className="flex space-x-2 justify-center mb-3"
    >
      <button
        className="inline-block px-6 py-2.5 bg-blue-600
 text-white font-medium text-xs leading-tight uppercase rounded shadow-md 
 hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none
 focus:ring-0 active:bg-blue-800 active:shadow-lg 
 transition duration-150 ease-in-out"
        data-testid="loginid"
        onClick={[Function]}
        type="button"
      >
        <span>
          Login
        </span>
      </button>
    </div>
  </div>
  <div
    className="mb-3"
  >
    <h1
      className="text-center"
    >
      Don't have an account ?
    </h1>
    <a
      href="#/register"
      onClick={[Function]}
    >
      <h1
        className="text-center text-blue-600 text-sm"
      >
        Register here
      </h1>
    </a>
  </div>
</div>
`;
